core/src/main/java/org/elasticsearch/common/rounding/Rounding.java

```diff
@@ -127,54 +127,59 @@ public byte id() {
 
         @Override
         public long round(long utcMillis) {
-            long rounded = field.roundFloor(utcMillis);
-            if (timeZone.isFixed() == false) {
-                // special cases for non-fixed time zones with dst transitions
-                if (timeZone.getOffset(utcMillis) != timeZone.getOffset(rounded)) {
-                    /*
-                     * the offset change indicates a dst transition. In some
-                     * edge cases this will result in a value that is not a
-                     * rounded value before the transition. We round again to
-                     * make sure we really return a rounded value. This will
-                     * have no effect in cases where we already had a valid
-                     * rounded value
-                     */
-                    rounded = field.roundFloor(rounded);
-                } else {
-                    /*
-                     * check if the current time instant is at a start of a DST
-                     * overlap by comparing the offset of the instant and the
-                     * previous millisecond. We want to detect negative offset
-                     * changes that result in an overlap
-                     */
-                    if (timeZone.getOffset(rounded) < timeZone.getOffset(rounded - 1)) {
-                        /*
-                         * we are rounding a date just after a DST overlap. if
-                         * the overlap is smaller than the time unit we are
-                         * rounding to, we want to add the overlapping part to
-                         * the following rounding interval
-                         */
-                        long previousRounded = field.roundFloor(rounded - 1);
-                        if (rounded - previousRounded < field.getDurationField().getUnitMillis()) {
-                            rounded = previousRounded;
-                        }
-                    }
+            while (true) {
+                long rounded = field.roundFloor(utcMillis);
+                long transitionTime = timeZone.previousTransition(utcMillis);
+                if (transitionTime == utcMillis) {
```

Comment:
I think this adds a `timeZone.previousTransition()` computation even for fixed time zone cases. I think to save computation we could return even earlier in unproblematic cases when `timeZone.isFixed()` like before.

Reply:
I'd rather have fewer branches unless the extra computation is onerous. [In this case I think it makes little difference.](https://github.com/JodaOrg/joda-time/blob/f08b3ea04843ef901f20de212ab53f1e183305c3/src/main/java/org/joda/time/tz/FixedDateTimeZone.java#L60-L70)